### DISCUSSION TITLE
working with serverless server

### DISCUSSION THREAD

**Discussion by xiaosha007** [taskforcesh/bullmq#654] [Q&A]
Hi, I am using a serverless server and implemented bullMQ as job scheduler on it. Each time i deployed a new version to my server, it will create a new server instance. In this situation, I noticed that the old server instance is not killed immediately and thus the worker is still serving with lower processing power. Therefore, there will be two workers serving at the same time, which is not what I desired. Do anyone has issue with this ? I am willing to learn and listen to your advice :D

**Comment by sabunt** [comment]
Hey @xiaosha007
Did you find solutions for your issue?
You can try something like
```
process.on("SIGINT", async () => {
  await worker.close();
  await queue.drain(); 
});

**Comment by manast** [comment]
I would not call ```queue.drain``` as that will remove jobs from the queue which is not often what you want to do when shutting down workers. Here some official information: https://docs.bullmq.io/guide/going-to-production#gracefully-shut-down-workers
